### PR TITLE
Reverted regression in cse, fixes issue #8964.

### PR DESCRIPTION
**Pull request by moorepants**

### CODE REVIEW

**Comment by moorepants** [comment]
@smichr @mihir-w

File: sympy/simplify/cse_main.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def postprocess_for_cse(expr, optimizations):
         The transformed expression.
     """
     if optimizations is None:
+        # FIXME : cse_optimizations is not defined anywhere.
```
Comment (smichr) [inline sympy/simplify/cse_main.py]: Should be `basic_optimizations` as defined above in the file, I believe.

File: sympy/simplify/cse_main.py [hunk 2]
```diff
@@ -442,19 +443,13 @@ def cse(exprs, symbols=None, optimizations=None, postprocess=None,
...
-        ImmutableSparseMatrix)
+    from sympy.matrices import (MatrixBase, Matrix, ImmutableMatrix,
+                                SparseMatrix, ImmutableSparseMatrix)
 
     # Handle the case if just one expression was passed.
-    if isinstance(exprs, Basic):
+    if isinstance(exprs, (Basic, MatrixBase)):
         exprs = [exprs]
 
-    is_singlematrix = False
```
Comment (smichr) [inline sympy/simplify/cse_main.py]: yes, you are right...we always want the result in a list. That was my fault.

**Comment by moorepants** [comment]
Did you fix the fixme?

**Comment by smichr** [comment]
No, I just wanted to get the most broken thing fixed. As I look at this more closely, it looks like those lines should be deleted as there is no default given for the pre and post processors. The None-handling is done in cse itself. If it's ok with you I can just fix that here and commit.

**Comment by moorepants** [comment]
Yeah, that's cool.

**Comment by smichr** [comment]
It's done.

**Comment by moorepants** [comment]
thanks

**Pull request merged**